src/api/Filter.js

```diff
@@ -0,0 +1,48 @@
+import { addExtremeValuesToArr } from '../utils/addExtremeValuesToArr';
+import { FilterTypes } from './FilterQueryBuilder';
+
+const filterFunctions = {
+  [FilterTypes.IN](filterValues, featureValue) {
+    return filterValues.includes(featureValue);
+  },
+  [FilterTypes.BETWEEN](filterValues, featureValue) {
+    const checkRange = (range) => {
+      const [lowerBound, upperBound] = range;
+      return featureValue >= lowerBound && featureValue <= upperBound;
+    };
+
+    return addExtremeValuesToArr(filterValues).some(checkRange);
+  }
+};
+
+export function filterApplicator({ filters = {}, returnedType = 'boolean' }) {
```

Comment:
As a developer `filter` could be a bit confused because of the JavaScript `filter` method. Furthermore, this way we will do:
```javacript
const filteredFeatures = features.filter(filter({ filters }));
```
I would leave it like this or rename it to `applyFilter`

Reply:
let's rename to applyFilter